src/main/java/BasicMAPF/Solvers/PIBT/PIBT_Solver.java

```diff
@@ -92,63 +111,116 @@ protected void init(MAPF_Instance instance, RunParameters parameters) {
+                        loopDetected = true;
+//                        for (Agent agent : this.agentPlans.keySet()) {
+//                            System.out.println("***************");
+//                            System.out.println(agent);
+//                            System.out.println("***************");
+//                            System.out.println("Agent's current priority: " + this.priorities.get(agent));
+////                            System.out.println(this.agentPlans.get(agent).getLastMove().prevLocation);
+////                            System.out.println(this.agentPlans.get(agent).getLastMove().currLocation);
+//                            System.out.println("timeStamp: " + (this.timeStamp-1));
+//                            System.out.println(this.agentPlans.get(agent).moveAt(this.timeStamp-1).prevLocation);
+//                            System.out.println(this.agentPlans.get(agent).moveAt(this.timeStamp-1).currLocation);
+//
+//                            System.out.println("Agent's status in prev identical timestamp: " + this.reachingGoalsConfigurations.get(i).get(agent));
+//                            System.out.println("timeStamp: " + (i-1));
+//                            System.out.println(this.agentPlans.get(agent).moveAt(i-1).prevLocation);
+//                            System.out.println(this.agentPlans.get(agent).moveAt(i-1).currLocation);
+//
+//                            System.out.println("-----------");
+//
+////                            System.out.println(this.agentPlans.get(agent).getLastMove().prevLocation);
+////                            System.out.println(this.agentPlans.get(agent).getLastMove().currLocation);
+//                            System.out.println("timeStamp: " + (this.timeStamp));
+//                            System.out.println(this.agentPlans.get(agent).moveAt(this.timeStamp).prevLocation);
+//                            System.out.println(this.agentPlans.get(agent).moveAt(this.timeStamp).currLocation);
+//
+//                            System.out.println("Agent's status in prev identical timestamp: " + this.reachingGoalsConfigurations.get(i).get(agent));
+//                            System.out.println("timeStamp: " + (i));
+//                            System.out.println(this.agentPlans.get(agent).moveAt(i).prevLocation);
+//                            System.out.println(this.agentPlans.get(agent).moveAt(i).currLocation);
+//
+//                            System.out.println("-----------");
+
+//                            if (agent.iD == 3 || agent.iD == 20) {
+//                                System.out.println(this.agentPlans.get(agent));
+//                            }
+//                        }
+
+                        break;
+                    }
+                }
+            }
+            if (loopDetected) {
+                System.out.println("LOOP DETECTED!!");
+                return null;
+            }
```

Comment:
So this loop is very slow. Instead, hold a set of location configurations (`Set<List<I_Location>>`) and clear it whenever an agent becomes "finished". If a new configuration already exists in the set, we have a loop. 
Then I think you will no longer need `reachingGoalsConfigurations`.